src/ImageSharp/Formats/Jpeg/Common/Decoder/JpegColorConverter.FromYCbCr.cs

```diff
@@ -121,63 +122,32 @@ internal static void ConvertCore(ComponentValues values, Span<Vector4> result)
                     tmp.MultiplyInplace(1.772F);
                     b.AddInplace(ref tmp);
 
-                    r.RoundAndDownscale();
-                    g.RoundAndDownscale();
-                    b.RoundAndDownscale();
+                    if (Vector<float>.Count == 4)
+                    {
+                        // TODO: Find a way to properly run & test this path on AVX2 PC-s! (Have I already mentioned that Vector<T> is terrible?)
+                        r.RoundAndDownscalePreAvx2();
```

Comment:
again I feel we should have a single call here called `.RoundAndDownscale()` that independently has the checks in. As long as `SimdUtils.IsAvx2CompatibleArchitecture` is a const then the jitter will remove the dead code for us but it will be much more readable as the callsite .

Reply:
But I feel that `SimdUtils.IsAvx2CompatibleArchitecture` would have to be an aggressively inlined getter only property for the jitter to follow the fact that the count is a jitter time constant and cause the dead code to be eliminated.

Reply:
I could be totally wrong and it's not the case, if its not then feel free to ignore.

Reply:
There are 2 solutions:
1. a static method taking 3 tuples by ref.
2. Defining a new struct for 3 tuples. 

Not sure if its worth to bother with this at the moment.

Reply:
If you don't think its worth the effort then feel free to leave it for now then.

Reply:
+Yeah the jitter tends to eliminate dead code afaik.